### ISSUE TITLE
Support --server option

### ISSUE THREAD

**Issue by unclechu** [daa84/neovim-gtk#21]
As in https://github.com/equalsraf/neovim-qt it would be cool to have this feature when you're able to start headless neovim-daemon and just connect to specific socket with GUI in any time, connect again you just closed GUI window and other useful stuff.

**Comment by unclechu** [comment]
Maybe additional info will help.

An example how to start "headless" neovim server:
```bash
env NVIM_LISTEN_ADDRESS="$HOME/.vim-server-socket-file-will-that-be-created" nvim --headless
```

An example how to connect to this neovim server with **neovim-qt** (can be attached and detached multiple times):
```bash
nvim-qt --server="$HOME/.vim-server-socket-file-will-that-be-created"
```

**Comment by unclechu** [comment]
I personally have this helper script to start neovim daemon:
https://github.com/unclechu/neovimrc/blob/40c6237f0ca30754f6254c19e7a7122aff32039d/apps/nvimd
And I start new daemon by command like this:
```bash
nvimd some-name
```
And then I can copy-paste socket path from this script output:
```text
Neovim server socket file path: /tmp/unclechu-nvim-some-name
Neovim binary: nvim
```

**Comment by daa84** [comment]
This became more complicated due to functionality changes. Gui file browser does not support remote connection. Also Some settings passed to nvim at start (g:GtkGuiLoaded, neovim runtime path and plugins list).

**Comment by unclechu** [comment]
Sad, but for me it would be okay to get it with extra features disabled (such as file browser) when working in this mode. I'm personally would be really happy to see this feature working.

**Comment by shaform** [comment]
+1 for this feature. I think the coolest part is that this enables us to connect to nvim sessions in remote servers via ssh. By doing so we are no longer being restricted by text-based terminal using using remote servers.

**Comment by unclechu** [comment]
I just found this commit: https://github.com/bkoropoff/neovim-gtk/commit/41b6a45eecf74cf0b4d44faa1e85512efcbd3f4e
It seems that person was trying to implement it (I personally haven't tested this yet).

**Comment by daa84** [comment]
yes, good starting point to implement server functionality. But some "local" logic was removed.

**Comment by unclechu** [comment]
I'm personally okay with that _local_ logic being removed since I personally don't use any of those features, only ligatures, font size changing, I turn off all the GUI panels. But I really need that external server feature. I really like ligatures support, but I really want my session to live when I accidentally close the window, so now I'm using neovim just inside tmux to make it work this way, giving up ligatures. I hope someday it will work in **neovim-gtk**, I just wish it would happen earlier than I find enough free time to implement it by myself.
